src/components/application_manager/src/commands/mobile/on_system_request_notification.cc

```diff
@@ -77,32 +77,29 @@ void OnSystemRequestNotification::Run() {
     return;
   }
 
-  if (RequestType::PROPRIETARY == request_type) {
-    /* According to requirements:
-       "If the requestType = PROPRIETARY, add to mobile API fileType = JSON
-        If the requestType = HTTP, add to mobile API fileType = BINARY"
-       Also in Genivi SDL we don't save the PT to file - we put it directly in
-       binary_data */
-
-    const std::string filename =
-        (*message_)[strings::msg_params][strings::file_name].asString();
+  const std::string& filename =
+      (*message_)[strings::msg_params][strings::file_name].asString();
+  if (!filename.empty()) {
     BinaryMessage binary_data;
     file_system::ReadBinaryFile(filename, binary_data);
+    if (RequestType::PROPRIETARY == request_type) {
+/* According to requirements:
+   "If the requestType = PROPRIETARY, add to mobile API fileType = JSON
+    If the requestType = HTTP, add to mobile API fileType = BINARY" */
 #if defined(PROPRIETARY_MODE)
-    AddHeader(binary_data);
+      AddHeader(binary_data);
 #endif  // PROPRIETARY_MODE
-
+      (*message_)[strings::msg_params][strings::file_type] = FileType::JSON;
+    } else if (RequestType::HTTP == request_type) {
+      (*message_)[strings::msg_params][strings::file_type] = FileType::BINARY;
+      if ((*message_)[strings::msg_params].keyExists(strings::url)) {
+        (*message_)[strings::msg_params][strings::timeout] =
+            policy_handler.TimeoutExchangeSec();
+      }
+    }
 #if defined(PROPRIETARY_MODE) || defined(EXTERNAL_PROPRIETARY_MODE)
     (*message_)[strings::params][strings::binary_data] = binary_data;
 #endif  // PROPRIETARY_MODE
```

Comment:
@LitvinenkoIra 
In case there is no file name binary data field is created anyway. Suggest to do like:
```
BinaryMessage data;
if (message.keyExist(filename)) {
   string filename = message[filename];
   ReadFile(filename, data);
   message[binary_data] = data;
}

if (Proprietary == requesttype) {
   #ifdef
      AddHeader(data)
   #endif
....
} else if (...) {
.....

```

Reply:
@dev-gh But binary data field is created inside `if (!filename.empty())` statement